crates/humanode-runtime/src/lib.rs

```diff
@@ -607,23 +629,59 @@ parameter_types! {
     pub WeightPerGas: Weight = Weight::from_ref_time(20_000);
 }
 
+impl pallet_evm_system::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type AccountId = EvmAccountId;
+    type Index = Index;
+    type AccountData = pallet_evm_balances::AccountData<Balance>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+}
+
+impl pallet_evm_balances::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type AccountId = EvmAccountId;
+    type Balance = Balance;
+    type ExistentialDeposit = ConstU128<500>;
+    type AccountStore = EvmSystem;
+    type DustRemoval = primitives_currency_swap_proxy::SwapUnbalanced<
+        currency_swap::EvmToNativeProxy,
+        TreasuryPot,
+        EvmBalancesPot,
+    >;
+}
+
+impl pallet_currency_swap::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type AccountIdTo = EvmAccountId;
+    type CurrencySwap = currency_swap::NativeToEvmOneToOne;
+    type WeightInfo = ();
+}
+
 impl pallet_evm::Config for Runtime {
-    type AccountProvider = pallet_evm::NativeSystemAccountProvider<Self>;
+    type AccountProvider = EvmSystem;
     type FeeCalculator = BaseFee;
     type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
     type WeightPerGas = WeightPerGas;
     type BlockHashMapping = pallet_ethereum::EthereumBlockHashMapping<Self>;
-    type CallOrigin = EnsureAddressTruncated;
-    type WithdrawOrigin = EnsureAddressTruncated;
-    type AddressMapping = HashedAddressMapping<BlakeTwo256>;
-    type Currency = Balances;
+    type CallOrigin = pallet_evm::EnsureAddressNever<EvmAccountId>;
+    type WithdrawOrigin = pallet_evm::EnsureAddressNever<EvmAccountId>;
+    type AddressMapping = pallet_evm::IdentityAddressMapping;
+    type Currency = EvmBalances;
     type RuntimeEvent = RuntimeEvent;
     type Runner = pallet_evm::runner::stack::Runner<Self>;
     type PrecompilesType = FrontierPrecompiles<Self>;
     type PrecompilesValue = PrecompilesValue;
     type ChainId = EthereumChainId;
     type BlockGasLimit = BlockGasLimit;
-    type OnChargeTransaction = fixed_supply::EvmTransactionCharger<Balances, FeesPot>;
+    type OnChargeTransaction = fixed_supply::EvmTransactionCharger<
+        EvmBalances,
+        primitives_currency_swap_proxy::SwapUnbalanced<
+            currency_swap::EvmToNativeProxy,
+            FeesPot,
+            EvmBalancesPot,
+        >,
```

Comment:
Move to the `currency_swap` mod.

Reply:
https://github.com/humanode-network/humanode/pull/673/commits/0507c03dba0fa004a5ea7e0fda1f654510967ca8